File: data/multilang_comparison/program_c/src/func10.c
#include "program_c.h" 
array_t* func10(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array4 = func11(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array4->id);
      free(params0.data);
      array4->refC--;
      if(array4->refC == 0) {
         free(array4->data);
         free(array4);
         DEBUG_FREE(array4->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array65 = func12(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array65->id);
      free(params0.data);
      array65->refC--;
      if(array65->refC == 0) {
         free(array65->data);
         free(array65);
         DEBUG_FREE(array65->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array101 = func13(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array101->id);
   free(params0.data);
   return array101;
}

File: data/multilang_comparison/program_nim/src/func10.nim
proc func10*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array4: ptr ArrayT = func11(addr params0, getPath(), loopsFactor)
      debugReturn(array4.id)
      dealloc(params0.data)
      dec array4.refC
      if array4.refC == 0:
         dealloc(array4.data)
         dealloc(array4)
         debugFree(array4.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array160: ptr ArrayT = func12(addr params0, getPath(), loopsFactor)
      debugReturn(array160.id)
      dealloc(params0.data)
      dec array160.refC
      if array160.refC == 0:
         dealloc(array160.data)
         dealloc(array160)
         debugFree(array160.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array197: ptr ArrayT = func13(addr params0, getPath(), loopsFactor)
   debugReturn(array197.id)
   dealloc(params0.data)
   return array197

